specification/configuration.md

```diff
@@ -89,6 +90,50 @@ environment variables:
   If `SPLUNK_CONFIG` is defined then these options are not required but MAY be
   defined.
 
+### Kubernetes
+
+While Kubernetes supports container technology that can be configured using
+environment variables, package management solutions such as Helm charts and
+Operators require YAML-based configuration. As a result, Kubernetes package
+management solutions MUST support the following YAML configuration options:
+
+- `splunkCommon`
+  - `clusterName` ()          : [REQUIRED] Name of the cluster.
+  - `distro` ()               : Which distribution of Kubernetes is deployed.
+  - `environment` ()          : Name of the environment; if not defined then skipped.
+  - `logsEngine` (`fluentd`)  : How to collect logs.
+  - `modes`
+    - `agent`                 : Deployed as a DaemonSet.
+      - `config` ()           : Updates configuration. Non-list options merged, list options override.
+    - `k8sClusterReceiver`    : Deployed as a standalone service.
+      - `config` ()           : Updates configuration. Non-list options merged, list options override.
+    - `gateway`               : Deployed as a clustered Service.
+      - `enabled` (`false`)   : Whether mode is deployed.
+      - `config` ()           : Updates configuration. Non-list options merged, list options override.
+  - `provider` ()             : Where Kubernetes is deployed.
+- `splunkObservability`
+  - `accessToken` ()          : [REQUIRED] Access token added to exported data.
+  - `logsEnabled` (`false`)   : Whether logs are collected and sent.
+  - `metricsEnabled` (`true`) : Whether metrics are collected, received, and sent.
+  - `realm` ()                : [REQUIRED] Realm configured for the exporter endpoint.
+  - `tracesEnabled` (`true`)  : Whether traces can be received and sent.
+- `splunkPlatform`
+  - `endpoint` ()             : [REQUIRED] Where to send exported data.
+  - `logsEnabled` (`false`)   : Whether logs are collected and sent.
+  - `metricsEnabled` (`true`) : Whether metrics are collected, received, and sent.
+  - `token` ()                : [REQUIRED] Token added to exported data.
+
+In addition, the following Kubernetes secret configuration options MUST be
+supported:
+
+- Splunk Observability
+  - `splunk_observability_access_token`
+- Splunk Platform
+  - `splunk_platform_hec_token`
+  - `hec_client_cert`
+  - `hec_client_key`
+  - `hec_ca_file`
```

Comment:
I think we need to add `splunk_platform_` prefix to all of these. Splunk Observability doesn't support ,them but they would be different anyway

Reply:
well LO supports hec... maybe just prefix with `splunk_`?

Reply:
LO supports hec, but can we assume that the same certificates always must be used for both backends?